### ISSUE TITLE
Audio stopped working on the host system

### ISSUE THREAD

**Issue by frkos** [home-assistant/plugin-audio#12]
Hello @pvizeli 
After the latest release [#9](https://github.com/home-assistant/audio/releases/tag/9) audio on my host system stopped working. Looks like hassio audio locked it...
Wben I run `aplay` I see this error:
`aplay: main:828: audio open error: Device or resource busy`
So I had to rollback to the version 8:
`ha audio update --version 8`
and it works again

**Comment by frkos** [comment]
Ohh.... it's upgraded back to 9 automatically =(

**Comment by pvizeli** [comment]
Okay, that will be the way to go. If someone read this and will contribute to the project, he can implement follow scenario:

- Add options to the Supervisor audio part disable audio claim
- Write the options to the audio container like add-on on start
- Pick this options on s6 init and modify the pulse files

**Comment by dimovnike** [comment]
is there a quck way to disable this? How to downgrade to an usable version?

**Comment by dimovnike** [comment]
where to find the ha binary? It's not present in any of the containers...

**Comment by dimovnike** [comment]
a temporary fix:

service hassio-supervisor stop
docker stop hassio_audio

Afte this, homeassistant stops working for about 3 minutes then it comes back and is usable.

**Comment by pvizeli** [comment]
Well, you can just load manual load the IDLE module for PulseAudio... just use google for the commands on PulseAudio

**Comment by jbirdjavi** [comment]
@pvizeli I'm running into this issue on Ubuntu 18.04.  I looked into loading `module-suspend-on-idle` for pulseaudio, but my `/etc/pulse/default.pa` file already has it loaded.  Is there somewhere else I should be looking?

**Comment by jbirdjavi** [comment]
@pvizeli What was the reasoning for https://github.com/home-assistant/audio/pull/11?  This has hosed two machines I'm running both home-assistant and mythtv on since the audio no longer works in mythtv.

**Comment by deisi** [comment]
@pvizeli 
I also don't understand what you mean by this. I think you mean I should configure pulse audio in the hassio_audio container, but if I do so using `docker run` changes will not be persistent. 

Is there anything I can change in `/usr/share/hassio/audio` to make hassio_audio not steal my host pc audio, as I currently don't want to have `hassio_audio`. This might change in the future, but as of right now, this is not sufficient for my use case. The only way that quickly worked is what @dimovnike mentioned. But turning off hassio_supervisor and hassio_audio is probably a bit much.

**Comment by deftdawg** [comment]
How do I permanently disable hassio_audio from running?   It breaks pulse audio on the host system. 

`journalctl --since "1 day ago" | grep pulseaudio` shows messages like these:
```
Mar 11 22:58:43 hades pulseaudio[1840]: [alsa-sink-ALC700 Analog] alsa-sink.c: Error opening PCM device front:0: Device or resource busy
Mar 11 22:58:43 hades pulseaudio[1840]: [alsa-sink-ALC700 Analog] alsa-sink.c: Error opening PCM device front:0: Device or resource busy
Mar 11 22:59:20 hades pulseaudio[1840]: [alsa-sink-ALC700 Analog] alsa-sink.c: Error opening PCM device front:0: Device or resource busy
Mar 11 22:59:27 hades pulseaudio[1840]: [alsa-sink-ALC700 Analog] alsa-sink.c: Error opening PCM device front:0: Device or resource busy
Mar 11 22:59:27 hades pulseaudio[1840]: [pulseaudio] sink-input.c: Failed to create sink input: sink is suspended.
```

Rebooting the host I can get audio back for a few minutes before these messages begin appearing again.

**Comment by pvizeli** [comment]
You need to use the `pactl` tool over the Terminal add-on to load the IDLE module if you need this.

The Supervisor is designed to get full control over your to host and everything is running as an add-on. If you use it differently, it ends up with such issues and they can get more in the future. You should maybe look into the OVA (OpenVirtualAppliance) or just use the Core installation.

**Comment by deisi** [comment]
I'm running HA on a x86 pc, that I also use as desktop and for various other things. I know that hassio is primarily run on raspberrys. Given that more of this will happen in the future, it might be a good time to switch to a virtual machine to host the hassio instance. I'm a bit worried about mapping all the USB devices, as I have about 20 of them or so, but it might eventually work.

**Comment by pvizeli** [comment]
That is the out of the box handling with supervisor, host -> supervisor -> add-ons. However, you can manage the PulseAudio instance by your self and tweak what do you want.

The Project have just 1 developer and that is me. If someone will jump into and maintain the generic installer with all issues and use case to make sure it work like you need, welcome to the team. But the size of the project and that is all maintain by just 1 person make that we need us to focus back to the roots. Home Assistant is a full manage system with allow to run the Supervisor on Debian/Ubuntu however, the nature of the Supervisor is the same.

**Comment by deisi** [comment]
> The Project have just 1 developer and that is me

You don't  need to justify anything to me. I'm very happy for your work. :) 

Putting hassio in a dedicated VM should be fine as well or am I wrong? This would also be future proof, has hassio can do in the VM what ever it likes. Plz correct me if I'm wrong as I would like to give that a try soon.

**Comment by deftdawg** [comment]
Per @pvizeli's clue, I believe these are the steps to get hassio audio to release audio back to the host: 

1. Select Supervisor tab from the sidebar
1. Install the `Terminal & SSH` add-on from the supervisor, start it, add it to the sidebar
1. Select the `Terminal` tab from the sidebar 
1. At the prompt type ``pactl load-module module-suspend-on-idle``

This should release the ALSA sink back to pulse audio on the host until the next _hassio_addon_ container restart.

**Comment by jbirdjavi** [comment]
@pvizeli I would still like to know the reasoning for #11 as it seems to be causing more issues than it's solving.

**Comment by dimovnike** [comment]
why is not module-suspend-on-idle loaded on default? Is there a reason for it? Or there are other addons that actually use pulse audio?

**Comment by jbirdjavi** [comment]
> However, you can manage the PulseAudio instance by your self and tweak what do you want.
This is only somewhat true, since a reboot of the host will reset things to default.  Is there a way to persist my changes?  I would prefer to stick with using the supervisor since there seems to be very little documentation for just installing and using Core in my brief searching.

**Comment by rothvaw** [comment]
I have the same problem, on my host system there is no sound and microphone. I'm not using any of audio steam/pass options in homeassistant, so I can live without it.
It's not working in my case:
```pactl load-module module-suspend-on-idle```

For my the best solution would be to turn off ```hassio_audio``` container from run by supervisor. Is there any option to do that?

**Comment by fourseven** [comment]
I’m in the same situation as the others - I run home-assistant through hass.io on a multipurpose x86 computer and suddenly lost sound on the host - it took a while to diagnose if I’m honest. I understand the intention of this container/project and will probably just build out a docker compose of all the things the hass.io install does (e.g. all the add-ons I use) currently but that’s a bigger project...

What worked for me (at least so far) was to stop and remove the audio container, as it looks like the supervisor gets a bit confused when it expects an image to be available locally and it isn’t.

```sh
# as root or prefixed with sudo:
docker stop hassio_audio
docker image rm -f homeassistant/amd64-hassio-audio:9
```

Super hacky but I couldn’t work out a way to tell the supervisor not to start the audio container, or a way to persist which pulse/alsa modules are loaded.

**Comment by Luxor73** [comment]
@fourseven 
Thank you, I removed hassio_audio image.
Before
`~ $ sudo fuser -fv /dev/snd/*
                     USER        PID ACCESS COMMAND
/dev/snd/controlC0:      root               373 f.... alsactl
                                    systemd-timesync   4229 F.... pulseaudio
/dev/snd/pcmC0D1p:   systemd-timesync   4229 F...m pulseaudio
/dev/snd/pcmC1D0c:   systemd-timesync   4229 F...m pulseaudio
`

Now
`~ $ sudo fuser -fv /dev/snd/*
                     USER        PID ACCESS COMMAND
/dev/snd/controlC0:  root        365 f.... alsactl
`

**Comment by deftdawg** [comment]
@fourseven I thinking restarting hassio_supervisor will just cause hassio_audio to be reinstalled again...  
Maybe if we can tag a dummy container as "homeassistant/amd64-hassio-audio:9" we can get around it that way.

**Comment by Luxor73** [comment]
@deftdawg Yes, after second reboot hassio_audio came back.
Maybe with a script or service on boot, something like:
sleep ...
hassio_audio = sudo docker ps | grep hassio_audio
if hassio_audio
docker stop hassio_audio
docker image rm -f homeassistant/amd64-hassio-audio:9
sudo reboot

**Comment by fourseven** [comment]
@deftdawg so:

```
docker pull homeassistant/amd64-hassio-audio:8
docker image tag homeassistant/amd64-hassio-audio:8 homeassistant/amd64-hassio-audio:9
```

😂

**Comment by patmann03** [comment]
Can we get a setting added so the host retains control over the audio?

I would prefer to keep using the supervisor as using the addon store is WAY more convenient than spinning up the docker containers yourself.

If there is a way to play audio files and do TTS directly from Home Assistant without needing access to the host system or Chromecast/Alexa I'm all ears.

**Comment by Daenara** [comment]
> @deftdawg so:
> 
> ```
> docker pull homeassistant/amd64-hassio-audio:8
> docker image tag homeassistant/amd64-hassio-audio:8 homeassistant/amd64-hassio-audio:9
> ```
> 
> 😂

I tried that and it worked fine through a few test restarts but this morning at 8AM it pulled the new version again...

**Comment by nickprotop** [comment]
> Can we get a setting added so the host retains control over the audio?
> 
> I would prefer to keep using the supervisor as using the addon store is WAY more convenient than spinning up the docker containers yourself.
> 
> If there is a way to play audio files and do TTS directly from Home Assistant without needing access to the host system or Chromecast/Alexa I'm all ears.

That would be perfect. Maybe expose the pulse server back to host?

**Comment by OPHoperHPO** [comment]
I developed a small add-on for hass.io that automatically fixes this problem (it executes the `pactl load-module module-suspend-on-idle` command) every time the container is restarted with the home assistant.

[Link](https://github.com/OPHoperHPO/hassio-addons/tree/master/pulseaudio_fix)

**Comment by jbirdjavi** [comment]
Thanks @OPHoperHPO!!  I still want to know why the module can't be loaded by default...

**Comment by rubenslook** [comment]
> I developed a small add-on for hass.io that automatically fixes this problem (it executes the `pactl load-module module-suspend-on-idle` command) every time the container with the home assistant is restarted.
> 
> [Link](https://github.com/OPHoperHPO/hassio-addons/tree/master/pulseaudio_fix)

Can you explain to me how to do the installation? I tried it through Hacs, but when I try to add as a custom repository an error appears. Thanks.

**Comment by jbirdjavi** [comment]
@rubenslook Use https://github.com/OPHoperHPO/hassio-addons as your custom repository.

**Comment by rubenslook** [comment]
> @rubenslook Use https://github.com/OPHoperHPO/hassio-addons as your custom repository.

![hacs](https://user-images.githubusercontent.com/20096032/77829347-43397880-7100-11ea-93c9-bd833488cb1b.jpg)

**Comment by jbirdjavi** [comment]
I don't recognize that screen at all.  What version of HA/supervisor are you running?

**Comment by rubenslook** [comment]
I'm sorry, I found it.

![hacs2](https://user-images.githubusercontent.com/20096032/77829665-40d81e00-7102-11ea-9d6f-be9cc1582f7c.jpg)

**Comment by deam0n** [comment]
> I developed a small add-on for hass.io that automatically fixes this problem (it executes the `pactl load-module module-suspend-on-idle` command) every time the container with the home assistant is restarted.
> 
> [Link](https://github.com/OPHoperHPO/hassio-addons/tree/master/pulseaudio_fix)

Thanks for this!! 

Struggling with this issue as well..

**Comment by MicroSur** [comment]
> I don't recognize that screen at all. 
It was HACS screen, everyone knows this )

**Comment by whitesabv** [comment]
Thank you so much, I’ve been looking for a solution to this problem for a long time.

**Comment by jgpazos70** [comment]
What I did:

Edit /etc/pulse/default.pa
   sudo nano /etc/pulse/default.pa
Comment line where module-suspend-on-idle is loaded:
   #load-module module-suspend-on-idle
Force load at boot of HDMI sink (is what I need)
   load-module module-alsa-sink device=hw:0,3

After rebooting HDMI can be selected in Settings-Sound so I have to change Dummy Sound and select HDMI. Also I have to select HDMI Configuration for Output 5.1 for make it work,

Even if hassio_audio is running Audio is connected to HDMI. If you experience problems in startup I think stopping hassio_audio just once make HDMI work until next reboot at least.

Regards.

**Comment by jwk4946** [comment]
> I developed a small add-on for hass.io that automatically fixes this problem (it executes the `pactl load-module module-suspend-on-idle` command) every time the container with the home assistant is restarted.
> 
> [Link](https://github.com/OPHoperHPO/hassio-addons/tree/master/pulseaudio_fix)

Thanks so much for this @OPHoperHPO ! I had been troubleshooting for hours before i figured out that it was caused by the audio container - your add on fixed the problem and saved me a huge headache

**Comment by Zireael** [comment]
Please make the hassio_audio service optional and [OFF] by default, as it causes unexpected changes to the host system (unexplained loss of audio playback).

Please add warning about potentially losing audio from the system if this service is enabled.

Please change docker restart policy for this container form 'always' to 'unless stopped'

This is very annoying and wasted a lot of time when people have to troubleshoot sudden problem with no audio in their system.

**Comment by gotschi** [comment]
I also would appreciate the hassio_audio service to be optional,
even though I run on odroid (which is supported) this creates trouble with permissions.
Even though I don't even need audio on this host and HA, it loop creates the container,
crashes it and restarts again.
Not ideal for any SD card.

**Comment by deftdawg** [comment]
Is use case for host audio via hassio_audio more edge case than running hassio in a non-dedicated Docker host?

I'm not arguing that most users probably deploy to a dedicated host like an RPi, but rather what proportion of users actually plug a speaker into the audio jack (or make use of sound over HDMI without running a desktop) and make use of it?

Is there an integration or scenario where it's commonly used?  I'm genuinely curious.

**Comment by Zireael** [comment]
I think it's used for voice commands ("Hassio, set my se* dungeon lights to blood red!").
But If I have audio output from RPI, I'd use it for movies/youtube/spotify rather than voice commands. But I suppose it's the opposite for other people.

**Comment by crzg** [comment]
In case this helps someone else - I spent all day trying to figure this out and have found success in changing 

`; autospawn = yes`

to 

`; autospawn = no`

In the following file: /etc/pulse/client.conf

Hassio is running through Docker on Ubuntu 20.04

**Comment by felixsanz** [comment]
i've been without audio for 3 months because i couldn't debug the problem and i didn't know it was the fault of this shitty container!!!!!! why take system audio???? docker is supposed to exist to separate things from system not to break the system. holy F!!!

**Comment by frenck** [comment]
@felixsanz Home Assistant is not just a Docker container. If you would like the more isolated approach on a pure containerized system, please use Home Assistant Core, which is available as just a Docker image.

Furthermore, please note, we have a Code of Conduct. There is no need to use those last words in your comments. Thanks 👍

**Comment by jbirdjavi** [comment]
@frenck There's a lot of frustration in this thread because no one has answered the question of why the audio container takes over the host audio.  It's been asked multiple times, and the pull request which made the change doesn't say why it was needed either.  Can someone answer this?

**Comment by frenck** [comment]
Home Assistant is designed as an appliance, it assumes it has the whole system to its own. If you want full control, use Home Assistant Core.

**Comment by jbirdjavi** [comment]
@frenck I understand that, but what was broken before that caused the need to remove the suspend-on-idle module in #11?

**Comment by frenck** [comment]
Popping sound when the audio channel was used on some single-board systems.

**Comment by deftdawg** [comment]
Popping on the input side messing up captures?  Or popping on output during playback?

**Comment by frenck** [comment]
Playback for sure, haven't tested the input myself.

However, it is not relevant, as this has been solved.

**Comment by felixsanz** [comment]
> Home Assistant is designed as an appliance, it assumes it has the whole system to its own. If you want full control, use Home Assistant Core.

who assumes that? you?

**Comment by frenck** [comment]
The project does, @felixsanz. I sense a lot of frustration on your end, and not sure why that comment was personal towards me. I found it a bit offending, to be honest.

Again, if you like to use your system for multiple things, please use Home Assistant Core, or use a Hypervisor to isolate.

**Comment by deftdawg** [comment]
It seems like the best course of action is to put some warnings on the project page or in the documentation and then close this as "Won't Fix", as the scenario where it occurs is usage that is "out of spec".

**Comment by frenck** [comment]
The documentation and clear project scopes is something that definitely needs addressing, as has been mentioned in this blog post:

<https://www.home-assistant.io/blog/2020/05/09/deprecating-home-assistant-supervised-on-generic-linux/>

Currently, I'm working on a full spec together with some others. Once that has been done and approved it will be published.

**Comment by felixsanz** [comment]
> The project does, @felixsanz. I sense a lot of frustration on your end, and not sure why that comment was personal towards me. I found it a bit offending, to be honest (most likely to caused by your previous choice of words, so that might be wrong interpreted from my end now).
> 
> Again, if you like to use your system for multiple things, please use Home Assistant Core, or use a Hypervisor to isolate.

obviously, i've been without sound on my main PC for 3 months and i've wasted more than 30 hours trying to debug everything, including downgrades of several packages like udev, kernels, etc. because i did not know that this container was messing up my system of out docker, of course that i'm mad and frustrated. this should be advised somewhere. you can't take system audio and don't even mention it on the website

**Comment by frenck** [comment]
> docker is ISOLATION. you want us to respect the philosophy of HA, but you don't respect the one from docker? a container is a container. i've never seen ship containers driving the ship to the nearest iceberg

I think you are mixing up 2 different things here @felixsanz. If you want to run Home Assistant on Docker, we provide Docker images: Home Assistant Core.

The solution you are using is an appliance that leverages Docker, but is not bound by it. It is just one of the pieces the appliance uses. For example, it communicates and manages the host system as well, via D-Bus.

**Comment by patmann03** [comment]
@felixsanz install this addon.  It resolves it.

https://github.com/OPHoperHPO/hassio-addons

**Comment by gotschi** [comment]
I respect the Supervised way of HA and the calm reactions from @frenck but cannot find a reason that this container should be running all the time on every system,
a option to make audio takeover optionally happen - and stop spinning up this container would be very appreciated especially since (I think) the majority doesn't even use audio output on SBCs.
Thanks

**Comment by frenck** [comment]
It provides a dummy audio output in those cases, to ensure an audio device can be expected.

**Comment by bilak** [comment]
well well, after 3 weeks trying to solve problem with QNAP I've found this. It really should be optional to run hassio_audio container.

**Comment by gotschi** [comment]
Please correct me if I'm wrong but incase there is no audio output device available -
shouldn't /dev/null suffice as audio dummy output? No need for a extra dummy container

**Comment by chozandrias76** [comment]
> @felixsanz install this addon. It resolves it.
> 
> https://github.com/OPHoperHPO/hassio-addons

I tried installing your addon through Home Assistant and it failed. I couldn't find the logs that said why, but I was following my guide here: https://community.matrix.one/t/home-assistant-zigbee-integration-with-matrix-creator/3003

**Comment by OPHoperHPO** [comment]
> > @felixsanz install this addon. It resolves it.
> > https://github.com/OPHoperHPO/hassio-addons
> 
> I tried installing your addon through Home Assistant and it failed. I couldn't find the logs that said why, but I was following my guide here: https://community.matrix.one/t/home-assistant-zigbee-integration-with-matrix-creator/3003
I updated the add-on to version 3.1, try updating the information about the add-ons and installing it again.

**Comment by TheRealBadRobot** [comment]
The Alsa & PulseAudio Fix always stops, has anyone met with this issue?

Log:
`[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] done.
[services.d] starting services
[services.d] done.
14
[cmd] pactl exited 0
[cont-finish.d] executing container finish scripts...
[cont-finish.d] done.
[s6-finish] waiting for services.
[s6-finish] sending all processes the TERM signal.`

**Comment by frenck** [comment]
ADR-0014: Installation method: Home Assistant Supervised

Has been finalized, approved and instated. It defines what is supported and what not, in this case, for running Home Assistant Supervised.

<https://github.com/home-assistant/architecture/blob/master/adr/0014-home-assistant-supervised.md>

More about what a supported installation method is:

<https://github.com/home-assistant/architecture/blob/master/adr/0012-define-supported-installation-method.md>

ADR-0014 states:

- The operating system is dedicated to running Home Assistant Supervised.
- No additional software, outside of the Home Assistant ecosystem, is installed.

This means this issue describes a use case that is not supported. Nevertheless, we are open for contributions that improve compatibility with a community-supported method like described in ADR-0012.

Closing this issue as a result of the above.

**Comment by jendib** [comment]
I wasted way too much time on this, thanks for reporting this bug. It's still beyond me how a Docker container breaking the audio of the host is considered normal behavior, but the fix addon from @OPHoperHPO is working nicely.

**Comment by dimovnike** [comment]
looks like @OPHoperHPO fix stopped working... byt @Daenara trick (with tagging) is working.

**Comment by jendib** [comment]
OPHoperHPO apparently fixed its addon and it's fine on my side at least (I can see that the module is loaded in logs).

**Comment by OPHoperHPO** [comment]
> looks like @OPHoperHPO fix stopped working... but @Daenara trick (with tagging) is working.

Open a new problem in the [repository](https://github.com/OPHoperHPO/hassio-addons) with the addon by attaching the addon's log there, also write the version of the addon that you use on your system, the hassio version and the hassio_audio container version.

**Comment by GazAt105** [comment]
@OPHoperHPO fix is not working for me and I have raised an issue in the [repository ](https://github.com/OPHoperHPO/hassio-addons/issues/1) covering same.
@Daenara trick is working for me.

**Comment by GazAt105** [comment]
I spoke to soon. HA has pulled another version and now it is back to running pulseaudio and blocking access to other apps.

**Comment by GazAt105** [comment]
I was wrong, @OPHoperHPO addon is working ok, I just didn't know it.

**Comment by jmtatsch** [comment]
Months without audio on linux are finally over.
Thank you @OPHoperHPO

**Comment by ORi0N** [comment]
Thank you, thank you, thank you @OPHoperHPO for that addon fix. I have spent the last 4 days testing different pulseaudio configuration in order to resolve this problem. Only to discover that the s6-supervise kept re-starting the pulseaudio in system mode, that takes control over the audio. I only found this article after having read the following: https://askubuntu.com/questions/1279000/how-can-i-disable-pulseaudio-system-mode-and-use-the-normal-systemd-per-user-ser
Which finally gave me the clue that it's HA doing this :s

I must add I don't understand at all why a docker container should grant the sole access over the host audio. Isn't like isolation like the whole part of containerization ? Please make it a setting so other new users like me don't lose 4 days over something trivial as this.

So again, thank you @OPHoperHPO for your fix.

**Comment by jbirdjavi** [comment]
I agree in not understanding the decision.  The only answer that we've been given is that Supervised Home Assistant is designed to be the only thing running on a given machine.  But that doesn't *really* answer the question of why this specific decision needed to be made.  My guess is that they don't want people to run the supervised version anymore so making it harder for people to use is a good thing in their eyes.  I'm grateful for the work that they do, but I just don't understand this one at all.

**Comment by ORi0N** [comment]
I'm in the process of migrating from OpenHAB. And my docker host is running multiple other containers with 'home automation' services like OpenHABv2, OpenHABv3, zoneminder, snapcast, opentherm gateway, nginx...
But as soon as I start the HA container all other services lose access to audio. No more multi room audio (snapcast), no more OpenHAB where still 90% of my code resides as I'm hesitant to migrate when the audio is this unstable. Etc, etc ...

Dockers philosophy is a sound one. Don't fight it, embrace it. And both communities can 
'cross-pollinate' and learn from one another ;)))

**Comment by RedFucktion** [comment]
Thanks for the addon @OPHoperHPO !

**Comment by nikogusm** [comment]
hi all, the @OPHoperHPO fix is not working for me. I am running HA on a raspberry os in a docker container

this is a lot frustrating because I have another service on raspberry that is running for audio and the hassio_audio always takes over everything :(

**Comment by Lipown** [comment]
@nikogusm same here :/

**Comment by mightymergz** [comment]
After learning way more than I wanted to about PulseAudio I found a simple fix to this issue. 

In /etc/pulse/client.conf put
`default-server = unix:/usr/share/hassio/audio/external/pulse.sock`

The Home Assistant Supervisor starts the pulseaudio server _inside_ the docker container. (This is a very aggressive pulseaudio instance as it always takes over the sound and doesn't let them go. The supervisor is also watching closely to make sure its pulseaudio is always running so it is very difficult to kill.) 

The docker container exposes the pulseaudio unix socket from inside the container at /data/external/pulse.sock mounted to /usr/share/hassio/audio/external/pulse.sock. Outside of the container the pulseaudio system does not know where to find the socket provided for by the hassio pulseaudio. I think it was intended to still work automatically, maybe it does on debian, but not on raspbian, at least not mine. (see https://github.com/home-assistant/plugin-audio)

**Comment by simonemarin** [comment]
None of the proposed solutions worked for me unfortunately. RPI4 Raspbian and HA in Docker.

**Comment by simonemarin** [comment]
> 
> 
> After learning way more than I wanted to about PulseAudio I found a simple fix to this issue.
> 
> In /etc/pulse/client.conf put
> `default-server = unix:/usr/share/hassio/audio/external/pulse.sock`
> 
> The Home Assistant Supervisor starts the pulseaudio server _inside_ the docker container. (This is a very aggressive pulseaudio instance as it always takes over the sound and doesn't let them go. The supervisor is also watching closely to make sure its pulseaudio is always running so it is very difficult to kill.)
> 
> The docker container exposes the pulseaudio unix socket from inside the container at /data/external/pulse.sock mounted to /usr/share/hassio/audio/external/pulse.sock. Outside of the container the pulseaudio system does not know where to find the socket provided for by the hassio pulseaudio. I think it was intended to still work automatically, maybe it does on debian, but not on raspbian, at least not mine. (see https://github.com/home-assistant/plugin-audio)

In my installation the volume is mounted in another location 
/home/pi/Docker/Hassio/audio | /data
so I changed the settings in client.conf accordingly but the issue is still there.

**Comment by shanness** [comment]
Damn... Hasn't fixed it.  My audio was broken again until I went in and restarted the the hassio_audio_fix again..

So sick of this annoying problem.  Does anyone know how to at least force it to use my HDMI sink so it won't interfere with my sound system and headset?

**Comment by simonemarin** [comment]
This is what I get :
pi@RPI4:~ $ docker exec -it hassio_audio pactl list short sinks
0	auto_null	module-null-sink.c	s16le 2ch 44100Hz	SUSPENDED

**Comment by shanness** [comment]
I think I've got a fix finally for this nasty problem that keeps breaking my mic and sound outputs on the host system.

Basically instead of setting the default to hdmi (which turns off regularly), I now load module-null-sink and set the default sink/source in hassio-audio to the null sink/source.

And I've integrated it into this great system service wrapper that monitors the hassio-audio docker container and re-runs it on every restart.  Pretty simple instructions on how to set it up including a systemd service file.

Seems to be working perfectly, and gets HA to stay away from all actual audio devices.  I've put in a pull request to the forked repo, so perhaps use that instead of mine once/if merged.

https://github.com/shanness/HomeAssistant-PulseAudio-Disable

Feedback welcome.

**Comment by alanmilinovic** [comment]
> I think I've got a fix finally for this nasty problem that keeps breaking my mic and sound outputs on the host system (nasty problem when mic can't be accessed during meetings or gaming, and have people yelling "Hello" at me as I scramble to fix it). Was almost at the point I was going to give up and delete HA..
> 
> Basically instead of setting the default to hdmi (which turns off regularly), I now load `module-null-sink` and set the default sink/source in hassio-audio to the null sink/source, as well as unloading `module-switch-on-connect` and `module-switch-on-port-available`
> 
> And I've integrated it into this great system service wrapper that monitors the hassio-audio docker container and re-runs it on every restart. Pretty simple instructions on how to set it up including a systemd service file.
> 
> Seems to be working perfectly, and should be immune from breaking from changes to the hassio containers. It's keeping HA to stay away from all actual audio devices. I've put in a pull request to the forked repo, so perhaps use that instead of mine once/if merged.
> 
> https://github.com/shanness/HomeAssistant-PulseAudio-Disable
> 
> Feedback welcome, hope this solves it for others. It might be worth unloading the suspend module before loading it like the [addon does](https://github.com/OPHoperHPO/hassio-addons/blob/master/pulseaudio_fix/main.py) (I assume to make it suspend them all immediately). I'll keep an eye on my logging to see if it's needed, and post here if I update the script.

I tried setting things up with systemd but it is not working automatically, once the system boots up. I am getting "Connection refused" failure, but restarting the service manually works. I think this loop function is not looping or get stuck in the early stage after reboot. I have NUC and MX Linux.

**Comment by alanmilinovic** [comment]
Now it looks like it is not working at all. Running manually "module-suspend-on-idle" command is returning number 16 but still no sound at all. Any help?

**Comment by alanmilinovic** [comment]
Ok, I am moving forward.

As I have bluetooth device connected to my host, I had to disable bluetooth in hassio_audio container by adding following line to Shanness script.
`docker exec -i hassio_audio pactl unload-module module-bluetooth-discover`

I also autostart Kodi on my system and had to start service once Kodi is up and running.
`sudo systemctl start pa-suspend`

Not sure why script doesn't work if executed before Kodi starts.

**Comment by bogdanni** [comment]
I measured a power saving of 1W+ on my basement mini computer which runs a minimal Linux installation dedicated to HA. Using `powertop` I noticed hundreds of events/s from `pulseaudio`, which is not even installed on the host.

This did it for me, an interpolation of comments above:
`docker exec -it hassio_audio pactl load-module module-suspend-on-idle`

**Comment by shanness** [comment]
> I tried setting things up with systemd but it is not working automatically, once the system boots up. I am getting "Connection refused" failure, but restarting the service manually works. I think this loop function is not looping or get stuck in the early stage after reboot. I have NUC and MX Linux.

Ahh, thanks for pointing that out.  I hadn't rebooted when I tried it.  I've fixed it now I think.  Put a sleep in after killing pulse in the docker container.  The problem was pulse hadn't restarted when my additions were run.

I was having the same problem too, but hadn't got around to debugging it.  Seems fixed now for me @alanmilinovic

**Comment by alanmilinovic** [comment]
> > I tried setting things up with systemd but it is not working automatically, once the system boots up. I am getting "Connection refused" failure, but restarting the service manually works. I think this loop function is not looping or get stuck in the early stage after reboot. I have NUC and MX Linux.
> 
> Ahh, thanks for pointing that out. I hadn't rebooted when I tried it. I've fixed it now I think. Put a sleep in after killing pulse in the docker container. The problem was pulse hadn't restarted when my additions were run.
> 
> I was having the same problem too, but hadn't got around to debugging it. Seems fixed now for me @alanmilinovic

Just tested but the result is same in my case. As soon as Kodi starts it break the audio so I put sleep for 10 seconds to let Kodi initialize and then I restart the pa-suspend service.

**Comment by fleXible** [comment]
@alanmilinovic Thanks to e-mail notification, I came back here just in time to take the bluetooth commands. Thx a lot guys. Though it's amazing what an endless timesink this has been but the HA teams is still not willing to put in a switch for everyone interested...

**Comment by shanness** [comment]
> Just tested and it is working now great, thank you.

Great!  good to hear.  It's been a nightmare for me until I figured out the null sink/source could be used! I'm actually moving to putting hassio on a VM (found a HP Proliant ML350 G6 in my apartments recycling :).

> What I would also add to your code is disabling bluetooth module in container, others had also issue with bluetooth. If you agree I can create a pull request?

Sure mate, go ahead.  I'd suggest putting a config option for it at the top to control if it gets used like I did.  Then I'll merge it and it will become part of my pull request.

**Comment by chatziko** [comment]
Here's a nice workaround for anyone who's interested. If in the host system you are using only alsa, you can instruct HA's pulse to ignore a certain card via udev. Just create a file `/etc/udev/rules.d/89-pulseaudio.rules` with the following content:

```
DRIVERS=="bcm2835_audio", ENV{PULSE_IGNORE}="1"
```

If needed, replace "bcm2835_audio" with the driver of your card, you can find it with
```
udevadm info -a -p `udevadm info -q path -n /dev/snd/controlC0`
```
(or `Control1`, etc).

This workaround doesn't involver any changes in HA, so it should persist updates. HA's pulse will run happily with any remaining cards in the system (or no card at all), and you can still use alsa in the host.

**Comment by piotrret** [comment]
This is probably the only good solution. On my laptop it works fine with the integrated intel card. I've been looking for this for hours and finally found it. Thanks a lot

**Comment by nehasjains** [comment]
I installed Homeassistant supervised on raspberrypi 4 running on debain 11

Please guide how to solve the following issues:-

How do I autostart pulseaudio and connect to bluetooth? I have already installed pulseaudio and I am able to connect to bluetooth speaker manually via the terminal.
It is possible to autostart pulseaudio and connect to bluetooth speaker automatically once the homeassistant is started?

How can I change the default media path?
Kindly guide

**Comment by DivanX10** [comment]
> В /etc/pulse/client.conf поместите
> `default-server = unix:/usr/share/hassio/audio/external/pulse.sock`

In this case, the microphone in the bluetooth headset does not work. In fact, to output audio to a bluetooth headset, it doesn't matter whether we use `default-server = unix:/usr/share/hassio/audio/external/pulse.sock` or `default-server =` . The sound goes to the bluetooth headset anyway, but the microphone on debian 11 works if instead `default-server = unix:/usr/share/hassio/audio/external/pulse.sock`, we indicate how it was by default there is `default-server =` I can't throw the bluetooth microphone in the Home Assistant. I have described in detail here about [my problem](https://community.home-assistant.io/t/should-i-go-there-whole-house-audio-and-then-some/39584/120?u=divanx10) and what I did. I do not know how to overcome this problem.

**Comment by gimmik84** [comment]
> Ok, I am moving forward.
> 
> As I have bluetooth device connected to my host, I had to disable bluetooth in hassio_audio container by adding following line to Shanness script. `docker exec -i hassio_audio pactl unload-module module-bluetooth-discover`
> 
> I also autostart Kodi on my system and had to start service once Kodi is up and running. `sudo systemctl start pa-suspend`
> 
> Not sure why script doesn't work if executed before Kodi starts.

In these days I had a similar problem with Bluetooth on my host Raspberry Pi OS with a Supervised Home Assistant. The audio was sent over Bluetooth from host to an Alexa dot speaker but after a couple of minutes of clear audio it was just crackling and annoying sounds. 
Without home assistant everything was fine.  

Since I did not need the bluetooth in home assistant I disabled the bluetooth integration and removed the Raspberry pi bluetooth entity/device from inside home assistant using the gui. Now Home assistant can not use the bluetooth but my host bluetooth and bluetooth speaker work fine.  I tried the Plugin (https://github.com/shanness/HomeAssistant-PulseAudio-Disable) but it did now work.
